src/main/java/org/broadinstitute/hellbender/tools/exome/allelefraction/AFModel.java

```diff
@@ -0,0 +1,239 @@
+        return IntStream.range(0, data.numSegments()).mapToDouble(s -> {
+            final AFParameters parameters = state.getParameters(s);
+            parameters.setPOutlier(pOutlier);
+            return segmentLogLikelihood(parameters, data.countsInSegment(s));
+        }).sum();
+    }
+
+    // the log likelihood of all segments, with modified mean bias
+    public static double logLikelihoodWithProposedMeanBias(final AFState state, final AFData data, final double meanBias) {
+        return IntStream.range(0, data.numSegments()).mapToDouble(s -> {
+            final AFParameters parameters = state.getParameters(s);
+            parameters.setmeanBias(meanBias);
+            return segmentLogLikelihood(parameters, data.countsInSegment(s));
+        }).sum();
+    }
+
+    // the log likelihood of all segments, with modified mean bias
+    public static double logLikelihoodWithProposedBiasRate(final AFState state, final AFData data, final double biasRate) {
+        return IntStream.range(0, data.numSegments()).mapToDouble(s -> {
+            final AFParameters parameters = state.getParameters(s);
+            parameters.setBiasRate(biasRate);
+            return segmentLogLikelihood(parameters, data.countsInSegment(s));
+        }).sum();
+    }
+
+    /**
+     * Compute the kth moment of allelic bias with respect to the het likelihood integrand, that is
+     * int [lambda^k * integrand d lambda]/int [integrand d lambda]
+     * We use the fact that int lambda^k*integrand = integrand(alpha -> alpha + k) * (alpha)*(alpha+1). . .(alpha + k - 1)/beta^k
+     *
+     * @param parameters allele fraction parameters at this het site
+     * @param count AllelicCount of alt and ref reads
+     * @param k the desired moment
+     * @return
+     */
+    public static double biasPosteriorMoment(final AFParameters parameters, final AllelicCount count, final AFIndicator indicator, final int k) {
+        final double beta = parameters.getBiasRate();
+        final double alpha = parameters.getMeanBias() * beta;
+        final double logNumerator = hetLogLikelihood(parameters.withIncrementedAlpha(k), count, indicator) + Gamma.logGamma(alpha + k) - (alpha + k)*log(beta);
+        final double logDenominator = hetLogLikelihood(parameters, count, indicator) + Gamma.logGamma(alpha) - alpha*log(beta);
+        return Math.exp(logNumerator - logDenominator);
+    }
+
+    /**
+     * Adds {@code numSamples - numBurnIn} Markov-Chain Monte-Carlo samples of the parameter posteriors (generated using
+     * Gibbs sampling) to the collections held internally.  The current {@link AFState} held internally is used
+     * to initialize the Markov Chain.
+     * @param numSamples    total number of samples per posterior
+     * @param numBurnIn     number of burn-in samples to discard
+     */
+    public void fitMCMC(final int numSamples, final int numBurnIn) {
```

Comment:
note that the definition of numSamples here doesn't match that in the documentation---you're adding numSamples, whereas elsewhere we add numSamples - numBurnIn. we can change it elsewhere if desired, but we should be consistent.